conf.d/java.d/jmx.conf

```diff
@@ -381,6 +381,81 @@
                     "name": "used"
                 }
             ]
-        }
+        },
+        // ####################################################################
+        // JVM ThreadMXBean dynamic charts with threads as dimensions
+        // ####################################################################
+        {
+            "id": "thread_cpu",
+            "title": "Thread total CPU utilization",
+            "units": "percentage",
+            "priority": "8100",
+            "chartType": "AREA",
+            "dimensionAlgorithm": "INCREMENTAL",
+            // dimensions are not predefined
+            "class": "dynamic",
```

Comment:
I am entirely happy with the solution used for disambiguation.

When it comes to config validation, I'd leave as much as possible to checks provided by type safety (versus using only code to validate optional/mutually exclusive fields).

I toyed with the idea of using [JsonTypeResolver](https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/annotation/JsonTypeResolver.html) instead of an extra property (here: `class`), but -- given my limited amount of Jackson-foo -- couldn't find a way to use it without adding lots of boilerplate code.

For starters, I'll rename the property to `dimensonType=fixed|dynamic`.